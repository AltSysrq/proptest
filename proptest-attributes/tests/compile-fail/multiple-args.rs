use proptest::{strategy::Strategy, string::string_regex};
use proptest_attributes::proptest;

fn gen_date() -> impl Strategy<Value = String> {
    let expr = "[0-9]{4}-[0-9]{2}-[0-9]{2}";
    string_regex(expr).unwrap()
}

#[proptest(gen_date())]
fn parse_date(date: String, _second_arg: String) {}
