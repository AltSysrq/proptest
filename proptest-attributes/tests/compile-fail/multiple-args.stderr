error: The `#[proptest]` macro can only be applied to a function with a single argument.
  --> $DIR/multiple-args.rs:10:1
   |
10 | fn parse_date(date: String, _second_arg: String) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
